intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/plugin/PluginVendors.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+ */
+
+package com.jetbrains.plugin.structure.intellij.plugin
+
+const val CORE_PLUGIN_ID = "com.intellij"
+const val SPECIAL_IDEA_PLUGIN_ID = "IDEA CORE"
+const val VENDOR_JETBRAINS = "JetBrains"
+const val VENDOR_JETBRAINS_SRO = "JetBrains s.r.o."
+
+/**
+ * Plugin Vendors that identify JetBrains-related plugins.
+ *
+ * @see com.intellij.ide.plugins.PluginManagerCore#isDeveloperByJetBrains
+ */
+object PluginVendors {
+  fun isDevelopedByJetBrains(plugin: IdePlugin): Boolean {
+    return CORE_PLUGIN_ID == plugin.pluginId ||
```

Comment:
AFAIU we don't need the pluginId checks here, only `plugin.vendor`

Reply:
Generally, we discover internal API usage by checking the plugin ID (`com.intellij` or `IDEA CORE`) or the plugin vendor.

Reply:
`com.intellij` is reserved for the actual platform "plugin" ID, it must never be used in (3rd party) plugins

Reply:
Updated to check just the `vendor`, ignoring plugin ID completely.